pkg/controller/gc.go

```diff
@@ -0,0 +1,133 @@
+package controller
+
+import (
+	"github.com/Sirupsen/logrus"
+	k8sapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/types"
+)
+
+type gc struct {
+	// reference to parent controller
+	controller *Controller
+
+	logger *logrus.Entry
+
+	k8s *unversioned.Client
+	ns  string
+}
+
+// collectCluster collects resources that matches cluster lable, but
+// does not belong to the cluster with given clusterUID
+func (gc *gc) collectCluster(cluster string, clusterUID types.UID) error {
+	option := k8sapi.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"etcd_cluster": cluster,
+			"app":          "etcd",
+		}),
+	}
+
+	return gc.collectResources(option, map[types.UID]bool{clusterUID: true})
+}
+
+// fullyCollect collects resources that are created by the controller,
+// but does not belong to any running etcd clusters.
+func (gc *gc) fullyCollect() error {
+	clusters, _, err := gc.controller.getClustersFromTPR()
```

Comment:
List without RV will serve from etcd. Freshness will be ensured. We can improve this by serving from operator cache. But now, we do not really store the UID. We can improve this later.